apps/api/src/domain/models/BaseDomainModel.ts

```diff
@@ -1,25 +1,26 @@
 import cloneToPlainObject from '../../lib/utilities/cloneToPlainObject';
-import { PartialDTO } from '../../types/partial-dto';
+import { DeepPartial } from '../../types/DeepPartial';
+import { DTO } from '../../types/partial-dto';
 
-type ModelConstructor<T extends BaseDomainModel = BaseDomainModel> = new (dto: PartialDTO<T>) => T;
+type ModelConstructor<T extends BaseDomainModel = BaseDomainModel> = new (dto: DTO<T>) => T;
 
 /**
  * TODO consider doing this with a mixin(s). There may be cases
  * where we want this behaviour on a non domain-model class without the
  * inheritance baggage.
  *  */
 export default class BaseDomainModel {
-    toDTO<TModel extends BaseDomainModel>(this: TModel): PartialDTO<TModel> {
+    toDTO<TModel extends BaseDomainModel>(this: TModel): DTO<TModel> {
         const result = cloneToPlainObject(this);
 
         return result;
     }
 
     // This allows us to use our instances as immutable data structures
-    clone<T extends BaseDomainModel>(this: T, overrides: PartialDTO<T> = {}): T {
+    clone<T extends BaseDomainModel>(this: T, overrides?: DeepPartial<DTO<T>>): T {
```

Comment:
Do we really want deep partial here? Let's leave it for now but address this if it causes any problems.